introspection_response_writer.go

```diff
@@ -204,35 +205,46 @@ func (f *Fosite) WriteIntrospectionResponse(rw http.ResponseWriter, r Introspect
 		return
 	}
 
-	expiresAt := int64(0)
+	response := map[string]interface{}{
+		"active": true,
+	}
+
+	extraClaimsSession, ok := r.GetAccessRequester().GetSession().(ExtraClaimsSession)
+	if ok {
+		extraClaims := extraClaimsSession.GetExtraClaims()
+		if extraClaims != nil {
+			for name, value := range extraClaims {
+				if !reflect.ValueOf(value).IsZero() {
```

Comment:
If I am not mistaken, this would remove zero values. But what if we have a parameter which is e.g. an enum and has a valid number of `0`? Maybe we could check for not nil instead?

Reply:
I think enumerations will not encode to JSON well anyway, if they are numeric.

We can maybe check for empty strings and nil only.

We can also remove this check and be more careful what we get to here in the first place. This is why I changed `ToMap` functions to not add claims if values are zero.

Reply:
I meant enums of type int, so `0,1,2,3`. `acr_values` is such an enum which has a `0` value that must be set if requested.

Reply:
So yes, which of two options I listed you would prefer?

Reply:
I think removing the check is legit as long as we add an option to `ExtraClaims` to also delete a value! Otherwise I would check for `nil`.

Reply:
> to also delete a value!

What do you mean? Why would you be deleting a value? We just now established that `GetExtraClaims` should be allowed just to add non-standard claims. But then who else would be setting it to make deletion useful? You simply should not return value from `GetExtraClaims` if you do not want to set it. And we should make sure default implementation does not return such values.

Reply:
> What do you mean? Why would you be deleting a value

If your session pipeline is complex :) But maybe I'm getting ahead of myself. Let's just use all the values from the map and not do any check.